pkg/snapshot/tracker.go

```diff
@@ -0,0 +1,154 @@
+package snapshot
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/trie"
+
+	iter "github.com/vulcanize/go-eth-state-node-iterator"
+)
+
+type trackedIter struct {
```

Comment:
Do we have a sense of how much overhead this iteration tracking adds?

Reply:
It seems pretty negligible. I didn't see any consistent slowdown with 8, 16, 32 threads